pepper-apis/src/main/java/org/broadinstitute/ddp/service/WorkflowService.java

```diff
@@ -59,7 +69,12 @@ public WorkflowService(PexInterpreter interpreter) {
             }
         }
 
-        return Optional.ofNullable(next);
+        Optional<WorkflowState> nextState = Optional.ofNullable(next);
+        nextState.ifPresent(nextWfState -> {
+            createActivityInstanceIfMissing(handle, fromState, nextWfState, operatorGuid, userGuid);
```

Comment:
A bit of nitpicking, but have we considered just calling this method directly right before the return, instead of having to nest it inside the `ifPresent()`? It's just a plain old `if` and we can also pull up the type check as well:
```
if (next != null && next.type() == ACTIVITY) {
    create((ActivityState) next);
}
return Optional()
```

Reply:
Yes, thanks for suggesting it, sounds like a viable alternative. However, it seems to me that the gain in speed is infinitesimal while readability can become slightly worse. Also, since we already use `Optional`, makes sense to follow the same style. Given all that, I'd stick to the original code 😃